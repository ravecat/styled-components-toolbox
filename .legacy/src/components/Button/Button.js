import React from "react";
import styled, { css } from "styled-components";

import Icon from "../Icon";

import { theme } from "common/theme";
import BaseComponent from "components/BaseComponent";

const Button = ({ children, icon, ...rest }) => (
  <Wrapper as="button" {...rest}>
    <IconWrapper>{icon ? <Icon asset={icon} /> : null}</IconWrapper>
    {children}
  </Wrapper>
);

const Wrapper = styled(BaseComponent).attrs(({ type = "button" }) => ({
  type
}))`
  display: inline-block;
  height: 32px;
  padding: 0 15px;
  line-height: 30px;
  text-align: center;
  letter-spacing: normal;
  cursor: pointer;
  outline: 0;
  white-space: nowrap;
  color: ${theme.primaryColor};
  background: ${theme.normalColor};
  border: 1px solid ${theme.primaryColor};
  border-radius: 0;

  &:hover {
    color: ${theme.accentColor};
    border-color: ${theme.accentColor};
  }

  ${({ primary }) =>
    primary &&
    css`
      && {
        background: ${theme.primaryColor};
        color: ${theme.normalColor};
      }

      &:hover {
        background: ${theme.accentColor};
        color: ${theme.textColor};
      }
    `};

  ${({ dashed }) =>
    dashed &&
    css`
      && {
        background: ${theme.textColor};
        border-color: ${theme.dividerColor};
        border: 1px dashed ${theme.lightPrimaryColor};
        color: ${theme.dividerColor};
      }

      &:hover {
        border-color: ${theme.accentColor};
        color: ${theme.accentColor};
      }
    `};

  ${({ rounded }) =>
    rounded &&
    css`
      && {
        border-radius: 4px;
      }
    `};

  ${({ block }) =>
    block &&
    css`
      && {
        display: inline-block;
        width: 100%;
      }
    `};

  ${({ link }) =>
    link &&
    css`
      && {
        background-color: transparent;
        border: 0;
        color: ${theme.primaryColor};
        text-decoration: underline;
      }

      &:hover {
        color: ${theme.accentColor};
      }
    `};

  ${({ ghost }) =>
    ghost &&
    css`
      && {
        background: transparent;
      }
    `};

  ${({ disabled }) =>
    disabled &&
    css`
      && {
        cursor: default;
        background: ${theme.textColor};
        border-color: ${theme.dividerColor};
        color: ${theme.dividerColor};
      }
    `};
`;

const IconWrapper = styled.i`
  ${Icon} {
    width: 14px;
    height: 14px;
    margin-right: 8px;
    vertical-align: -1.75px;
  }
`;

export default Button;
