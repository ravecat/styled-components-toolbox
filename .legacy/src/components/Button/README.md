## Button

A basic button. Buttons are used as triggers for actions in forms, toolbars, dialog footers and as stand-alone action triggers.

### API

| Property   | Description                 | Type    | Default |
| ---------- | --------------------------- | ------- | ------- |
| _disabled_ | disabled state of button    | boolean | `false` |
| _href_     | redirect url of link button | string  | -       |

`<Button>Hello world!</Button>` will be rendered into `<button><span>Hello world!</span></button>`, and all the properties which are not listed above will be transferred to the `<button>` tag.

`<Button href="http://example.com">Hello world!</Button>` will be rendered into `<a href="http://example.com"><span>Hello world!</span></a>`
