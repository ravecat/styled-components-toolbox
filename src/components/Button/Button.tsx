import React from "react";
import styled, { css } from "styled-components";
import { BaseComponent, BaseProps } from "common/BaseComponent";

export interface ButtonProps extends BaseProps {
  icon?: React.ReactNode;
  type?: "button" | "submit" | "reset" | undefined;
  primary?: boolean;
  link?: boolean;
  ghost?: boolean;
  disabled?: boolean;
  block?: boolean;
  dashed?: boolean;
  rounded?: boolean;
  onClick?: () => void;
}

const Wrapper = styled(BaseComponent).attrs<ButtonProps>(
  ({ type = "button" }) => ({
    type,
  })
)<ButtonProps>`
  display: inline-block;
  height: 32px;
  padding: 0 15px;
  line-height: 30px;
  text-align: center;
  letter-spacing: normal;
  cursor: pointer;
  outline: 0;
  white-space: nowrap;
  color: ${({ theme }) => theme.neutralColor};
  border: 1px solid ${({ theme }) => theme.lightPrimaryColor};
  background: ${({ theme }) => theme.lightPrimaryColor};
  border-radius: 0;

  ${({ primary }) =>
    primary &&
    css`
      && {
        background: ${({ theme }) => theme.darkPrimaryColor};
        border: 1px solid ${({ theme }) => theme.darkPrimaryColor};
      }
    `};

  ${({ ghost }) =>
    ghost &&
    css`
      && {
        background: transparent;
        color: ${({ theme }) => theme.lightPrimaryColor};
      }
    `};

  ${({ dashed }) =>
    dashed &&
    css`
      && {
        background: ${({ theme }) => theme.neutralColor};
        border-color: ${({ theme }) => theme.dividerColor};
        border: 1px dashed ${({ theme }) => theme.lightPrimaryColor};
        color: ${({ theme }) => theme.dividerColor};
      }
    `};

  ${({ rounded }) =>
    rounded &&
    css`
      && {
        border-radius: 4px;
      }
    `};

  ${({ block }) =>
    block &&
    css`
      && {
        display: inline-block;
        width: 100%;
        color: ${({ theme }) => theme.lightPrimaryColor};
      }
    `};

  ${({ link }) =>
    link &&
    css`
      && {
        background-color: transparent;
        border: 1px solid transparent;
        color: ${({ theme }) => theme.lightPrimaryColor};
        text-decoration: underline;
      }
    `};

  ${({ disabled }) =>
    disabled &&
    css`
      && {
        cursor: default;
        background: ${({ theme }) => theme.neutralColor};
        border-color: ${({ theme }) => theme.dividerColor};
        color: ${({ theme }) => theme.dividerColor};
      }
    `};
`;

// const IconWrapper = styled.i`
//   ${Icon} {
//     width: 14px;
//     height: 14px;
//     margin-right: 8px;
//     vertical-align: -1.75px;
//   }
// `;

/**
 * Primary UI component for user interaction
 */
export const Button: React.FC<ButtonProps> = ({
  children,
  ...rest
}: ButtonProps) => (
  <Wrapper {...rest} as="button">
    {/* <IconWrapper>{icon ? <Icon asset={icon} /> : null}</IconWrapper> */}
    {children}
  </Wrapper>
);
