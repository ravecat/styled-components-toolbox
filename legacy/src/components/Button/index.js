import { renderComponent, branch } from "recompose";

import ButtonLink from "./components/ButtonLink";
import Button from "./Button";

export default branch(({ href }) => href, renderComponent(ButtonLink))(Button);
