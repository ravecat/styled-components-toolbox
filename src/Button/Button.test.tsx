import React from "react";
import renderer from "react-test-renderer";
import { Button } from "Button/Button";

it("renders correctly", () => {
  const tree = renderer
    .create(<Button backgroundColor="fff" label="Test" primary />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
