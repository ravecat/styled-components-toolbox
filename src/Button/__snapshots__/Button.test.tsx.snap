// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="storybook-button storybook-button--medium storybook-button--primary"
  style={
    Object {
      "backgroundColor": "fff",
    }
  }
  type="button"
>
  Test
</button>
`;
